frontend/src/app/modules/bim/ifc_models/ifc-viewer/ifc-viewer.component.ts

```diff
@@ -94,13 +94,15 @@ export class IFCViewerComponent implements OnInit, OnDestroy {
 
   @HostListener('mousedown')
   enableKeyBoard() {
-    this.keyboardEnabled = true;
-    this.ifcViewer.setKeyboardEnabled(true);
+    if (this.modelCount) {
```

Comment:
This is unrelated to this PR, but I noticed that an error could occur from this method when `ifcViewer` is not yet defined. Could you add a guard whether it is defined here?